test/libsolidity/semanticTests/functionTypes/pass_function_types_internally.sol

```diff
@@ -7,10 +7,9 @@ contract C {
         return x(a);
     }
 
-    function g(uint256 x) public returns (uint256) {
+    function g(uint256 x) public pure returns (uint256) {
```

Comment:
Why is this change here now?

Reply:
Something weird happens, without pure, it passes with compileViaYul in isoltest, but fails in soltest with --optimize (it triggers warning and somehow fails later on)

with pure, it doesn't pass viaYul even in isoltest

Reply:
Can you isolate this case and open a tracking issue?

Reply:
I can open issue, but I do not fully understand what does it mean to isolate test case?

Reply:
For the record: the change here causes IR codegen to generate invalid Yul, that's why enforce-via-yul starts to ignore this. Without this change, there was a Yul parser error that bubbled up as weird ICE error - but that should be provisionally fixed by https://github.com/ethereum/solidity/pull/8845 now.
However, we should investigate the invalid yul code generated, when the version *with* ``pure`` is compiled via yul.